subprojects/execution/src/main/java/org/gradle/internal/execution/UnitOfWork.java

```diff
@@ -55,7 +57,22 @@
 
     long markExecutionTime();
 
-    CacheHandler createCacheHandler();
+    /**
+     * Return a reason to disable caching for this work.
+     * When returning {@link Optional#empty()} if caching can still be disabled further down the pipeline.
+     */
+    Optional<CachingDisabledReason> shouldDisableCaching();
+
+    /**
+     * This is a temporary measure for Gradle tasks to track a legacy measurement of all input snapshotting together.
+     */
+    default void markSnapshottingInputsFinished(CachingState cachingState) {}
+
+    /**
+     * Is this work item allowed to load from the cache, or if we only allow it to be stored.
+     */
+    // TODO Make this part of CachingState instead
```

Comment:
Will you address this TODO as part of this PR?

Reply:
Wasn't planning to.